### PR TITLE
python3Packages.s3fs: 0.5.2 -> 2021.5.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 2021.5.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126046` at 844278b0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.caffe</li>
    <li>python38Packages.dask-gateway</li>
    <li>python38Packages.dask-jobqueue</li>
    <li>python38Packages.dask-ml</li>
    <li>python38Packages.dask-mpi</li>
    <li>python38Packages.dask-xgboost</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.streamz</li>
    <li>python38Packages.stumpy</li>
    <li>python39Packages.caffe</li>
    <li>python39Packages.dask-gateway</li>
    <li>python39Packages.dask-jobqueue</li>
    <li>python39Packages.dask-mpi</li>
    <li>python39Packages.dask-xgboost</li>
    <li>python39Packages.streamz</li>
  </ul>
</details>
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/417aa235a371a8648b7e9d12d9a2f3f8">python38Packages.dask-image</a></li>
    <li><a href="https://gist.github.com/3686f991290055e6afd4bae96bc3c816">python38Packages.gcsfs</a></li>
    <li><a href="https://gist.github.com/05e554c8da0e1e1b873b87214d23f76a">python38Packages.imgaug</a></li>
    <li><a href="https://gist.github.com/5c9e34b86277ff00ae93bc9f8206fd12">python38Packages.mask-rcnn</a></li>
    <li><a href="https://gist.github.com/317b22cc17d831cd2c8e52e85db49e2e">python38Packages.rl-coach</a></li>
    <li><a href="https://gist.github.com/fc6bd11db4c737360c3c01b63dc0215d">python38Packages.sunpy</a></li>
    <li><a href="https://gist.github.com/4de88fa3975691cd2542deeeeb32ed1b">python39Packages.dask-image</a></li>
    <li><a href="https://gist.github.com/2eab0f984f9020d00e5361e14c36b30d">python39Packages.gcsfs</a></li>
    <li><a href="https://gist.github.com/5d79536309438ec811ebeb6d89f28a1b">python39Packages.imgaug</a></li>
    <li><a href="https://gist.github.com/12a0c7562fc0428ce5eb39a3b0553aff">python39Packages.mask-rcnn</a></li>
    <li><a href="https://gist.github.com/2336b62253657a00973ee53f5ef0be9d">python39Packages.rl-coach</a></li>
    <li><a href="https://gist.github.com/4393f498c7c46a756c5a9edb23301bf4">python39Packages.sunpy</a></li>
  </ul>
</details>
<details>
  <summary>3 packages skipped due to time constraints:</summary>
  <ul>
    <li>python38Packages.clifford</li>
    <li>python38Packages.dask-glm</li>
    <li>python39Packages.dask-glm</li>
  </ul>
</details>
<details>
  <summary>51 packages built successfully:</summary>
  <ul>
    <li>python38Packages.aplpy</li>
    <li>python38Packages.asdf</li>
    <li>python38Packages.batchgenerators</li>
    <li>python38Packages.dask</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.fsspec</li>
    <li>python38Packages.glymur</li>
    <li>python38Packages.image-match</li>
    <li>python38Packages.imagecorruptions</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.pims</li>
    <li>python38Packages.pyfftw</li>
    <li>python38Packages.pyregion</li>
    <li>python38Packages.pyspnego</li>
    <li>python38Packages.pytest-astropy</li>
    <li>python38Packages.pytest-astropy-header</li>
    <li>python38Packages.reproject</li>
    <li>python38Packages.s3fs</li>
    <li>python38Packages.scikitimage</li>
    <li>python38Packages.slicedimage</li>
    <li>python38Packages.smbprotocol</li>
    <li>python38Packages.sparse</li>
    <li>python38Packages.spectral-cube</li>
    <li>python38Packages.stytra</li>
    <li>python38Packages.tensorly</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
    <li>python39Packages.aplpy</li>
    <li>python39Packages.asdf</li>
    <li>python39Packages.batchgenerators</li>
    <li>python39Packages.dask</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.fsspec</li>
    <li>python39Packages.glymur</li>
    <li>python39Packages.image-match</li>
    <li>python39Packages.imagecorruptions</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.pims</li>
    <li>python39Packages.pyfftw</li>
    <li>python39Packages.pyregion</li>
    <li>python39Packages.pyspnego</li>
    <li>python39Packages.pytest-astropy</li>
    <li>python39Packages.pytest-astropy-header</li>
    <li>python39Packages.reproject</li>
    <li>python39Packages.s3fs</li>
    <li>python39Packages.scikitimage</li>
    <li>python39Packages.slicedimage</li>
    <li>python39Packages.smbprotocol</li>
    <li>python39Packages.spectral-cube</li>
    <li>python39Packages.tokenizers</li>
    <li>python39Packages.transformers</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 126046` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.clifford</li>
<li>python38Packages.dask-jobqueue</li>
<li>python38Packages.dask-xgboost</li>
<li>python38Packages.rl-coach</li>
<li>python39Packages.dask-jobqueue</li>
<li>python39Packages.dask-xgboost</li>
<li>python39Packages.rl-coach</li>
</ul>
</details>
<details>
<summary>
85 packages built:
</summary>
<ul>
<li>lexicon</li>
<li>powerdns-admin</li>
<li>python38Packages.aplpy</li>
<li>python38Packages.asdf</li>
<li>python38Packages.batchgenerators</li>
<li>python38Packages.caffe</li>
<li>python38Packages.dask</li>
<li>python38Packages.dask-gateway</li>
<li>python38Packages.dask-glm</li>
<li>python38Packages.dask-image</li>
<li>python38Packages.dask-ml</li>
<li>python38Packages.dask-mpi</li>
<li>python38Packages.datasets</li>
<li>python38Packages.datashader</li>
<li>python38Packages.distributed</li>
<li>python38Packages.fsspec</li>
<li>python38Packages.gcsfs</li>
<li>python38Packages.glymur</li>
<li>python38Packages.image-match</li>
<li>python38Packages.imagecorruptions</li>
<li>python38Packages.imgaug</li>
<li>python38Packages.intake</li>
<li>python38Packages.mask-rcnn</li>
<li>python38Packages.pims</li>
<li>python38Packages.pyfftw</li>
<li>python38Packages.pyregion</li>
<li>python38Packages.pyspnego</li>
<li>python38Packages.pytest-astropy</li>
<li>python38Packages.pytest-astropy-header</li>
<li>python38Packages.python3-saml</li>
<li>python38Packages.reproject</li>
<li>python38Packages.s3fs</li>
<li>python38Packages.scikitimage</li>
<li>python38Packages.slicedimage</li>
<li>python38Packages.smbprotocol</li>
<li>python38Packages.spacy-transformers</li>
<li>python38Packages.sparse</li>
<li>python38Packages.spectral-cube</li>
<li>python38Packages.streamz</li>
<li>python38Packages.stumpy</li>
<li>python38Packages.stytra</li>
<li>python38Packages.sunpy</li>
<li>python38Packages.tensorly</li>
<li>python38Packages.tokenizers</li>
<li>python38Packages.transformers</li>
<li>python38Packages.xmlsec</li>
<li>python38Packages.zeep</li>
<li>python39Packages.aplpy</li>
<li>python39Packages.asdf</li>
<li>python39Packages.batchgenerators</li>
<li>python39Packages.caffe</li>
<li>python39Packages.dask</li>
<li>python39Packages.dask-gateway</li>
<li>python39Packages.dask-glm</li>
<li>python39Packages.dask-image</li>
<li>python39Packages.dask-mpi</li>
<li>python39Packages.datasets</li>
<li>python39Packages.distributed</li>
<li>python39Packages.fsspec</li>
<li>python39Packages.gcsfs</li>
<li>python39Packages.glymur</li>
<li>python39Packages.image-match</li>
<li>python39Packages.imagecorruptions</li>
<li>python39Packages.imgaug</li>
<li>python39Packages.intake</li>
<li>python39Packages.mask-rcnn</li>
<li>python39Packages.pims</li>
<li>python39Packages.pyfftw</li>
<li>python39Packages.pyregion</li>
<li>python39Packages.pyspnego</li>
<li>python39Packages.pytest-astropy</li>
<li>python39Packages.pytest-astropy-header</li>
<li>python39Packages.python3-saml</li>
<li>python39Packages.reproject</li>
<li>python39Packages.s3fs</li>
<li>python39Packages.scikitimage</li>
<li>python39Packages.slicedimage</li>
<li>python39Packages.smbprotocol</li>
<li>python39Packages.spectral-cube</li>
<li>python39Packages.streamz</li>
<li>python39Packages.sunpy</li>
<li>python39Packages.tokenizers</li>
<li>python39Packages.transformers</li>
<li>python39Packages.xmlsec</li>
<li>python39Packages.zeep</li>
</ul>
</details>

File: pkgs/development/python-modules/fsspec/default.nix [hunk 1]
```diff
@@ -43,8 +54,10 @@ buildPythonPackage rec {
     "test_touch"
   ];
 
+  pythonImportsCheck = [ "" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/fsspec/default.nix:57]: You missed something here.

Comment (fabaff) [reply pkgs/development/python-modules/fsspec/default.nix]: Thanks, fixed.
